### PR TITLE
Properly load async javascsript

### PR DESCRIPTION
**Pull request by benrudolph**
@biyeun this should handle any issues with async javascript loading. i've tested this all locally accept for the custom reports. @kkrampa / @lwyszomi could you just load these 3 reports and ensure there are no javascript errors? :tropical_fish: with `w=1`
- care_pathways: Table Card Report
- intrahealth: Tableau De Bord
- world_vision: Child Report

fyi: @dimagi/team-commcare-hq if you ever need to load javascript asynchronously, use `getScript`. pageload event does _not_ work when you add a script tag after the page has loaded (which is what report filters do). this bug was uncovered through using the CDN to grab static files

### CODE REVIEW

File: custom/common/templates/common/drilldown_options.html [hunk 1]
```diff
@@ -31,13 +31,11 @@
...
-<link href="{% static 'select2-3.4.5-legacy/select2.css' %}" rel="stylesheet"/>
-<script src="{% static 'select2-3.4.5-legacy/select2.min.js' %}"></script>
-
-<script type="text/javascript" src="{% static 'reports/ko/report_filter.drilldown_options.js' %}"></script>
-<script type="text/javascript" src="{% static 'common/ko/report_filter.drilldown_options.js' %}"></script>
 <script type="text/javascript">
-    $(function () {
+    $.when(
+        "{% static 'reports/ko/report_filter.drilldown_options.js' %}",
+        "{% static 'common/ko/report_filter.drilldown_options.js' %}",
```
Comment (dannyroberts) [inline custom/common/templates/common/drilldown_options.html]: I didn't know you could use when like this directly on strings to mean "load these resources"

**Comment by dannyroberts** [comment]
Looks like a great solution!

File: custom/common/templates/common/drilldown_options.html (hunk 1 above)
Comment (benrudolph) [inline custom/common/templates/common/drilldown_options.html]: haha you can't

**Comment by biyeun** [comment]
thanks ben, this will be better than the temp solution i quickly put up

File: custom/world_vision/templates/world_vision/location_filter.html [hunk 2]
```diff
@@ -23,12 +23,9 @@
 {% endblock %}
 
 {% block filter_js %}
-<link href="{% static 'select2-3.4.5-legacy/select2.css' %}" rel="stylesheet"/>
-<script src="{% static 'select2-3.4.5-legacy/select2.min.js' %}"></script>
-
-<script type="text/javascript" src="{% static 'world_vision/ko/report_filter.drilldown_options.js' %}"></script>
+<script type="text/javascript" src=></script>
```
Comment (biyeun) [inline custom/world_vision/templates/world_vision/location_filter.html]: src=""

**Pull request merged**